sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -293,6 +293,28 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  /**
+   * Create a partition specification map with filters.
+   */
+  override def visitDropPartitionSpec(
+      ctx: DropPartitionSpecContext): Seq[Expression] = {
+    withOrigin(ctx) {
+      ctx.dropPartitionVal().asScala.map { pFilter =>
+        if (pFilter.identifier() == null || pFilter.constant() == null ||
+            pFilter.comparisonOperator() == null) {
+          throw new ParseException(s"Invalid partition spec: ${pFilter.getText}", ctx)
```

Comment:
Can you add tests for this exception in `DDLParserSuite.scala`?

Reply:
sure, will do ASAP, thanks.